packages/core/src/utilities/worldToImageCoords.ts

```diff
@@ -0,0 +1,133 @@
+import { vec3, vec2, mat4, vec4 } from 'gl-matrix';
+import { metaData } from '..';
+import { Point2, Point3 } from '../types';
+
+type Options = {
+  imageId?: string;
+};
+
+/**
+ * Given the 3d coordinates on the world space, and options which includes the imageId
+ * it returns the image coordinates (IJ) on the image space. The image space is
+ * defined with [0,0] being on the top left corner of the first top left pixel,
+ * the [1,1] being on the bottom right corner of the first top left pixel.
+ * @param worldCoords - The 3d coordinates on the world.
+ * @param options - options which includes the imageId
+ * @returns The 2d coordinates on the image.
+ *
+ */
+function worldToImageCoords(
+  worldCoords: Point3,
```

Comment:
My only concern with this method is unlike the very simiple image to world, this calculates an entire matrix first, which is then re-usable.


If we call this for 1000 points, the matrix will be calculated again and again for each point.

Maybe both methods should instead take an array of points and map them all?

imageToWorldCoords would just be a loop, but worldToImageCoords could then re-use the matrix?